application/include/GameLogic.h

```diff
@@ -0,0 +1,26 @@
+#ifndef _GameLogic_H_
+#define _GameLogic_H_
+
+
+class GameLogic {
+  public:
+    GameLogic();
+    void update(float);
+    void handlePlayerDeath(int);
+    void togglePause();
+    void generateMap();
+
+
+  private:
+    // walrus 1: player
+    // Walrus 2: player
+    // stage: stage
+    //need to implement the above
+    // we also have a gamestate variable, but no gamestate class in the diagram. Do we need one? or should we use ints?
```

Comment:
Thanks for pointing this out - in pong I used a simple GameState enum datatype like: 
`enum GameState {mainMenu, playing, paused, gameOverMenu};`
I think this would be sufficient and maybe more explicit than just ints.